File: dagger2/src/main/java/com/laioffer/enjoy/demo1_入门/Demo1Activity.java
package com.laioffer.enjoy.demo1_入门;

import javax.inject.Inject;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @date 2022/10/3 9:44 下午
 */
public class Demo1Activity extends AppCompatActivity {

  @Inject
  Computer mComputer;

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    DaggerComputerComponent.create().injectMainActivity(this);
  }
}

File: my-mvx/src/main/java/com/laioffer/my_mvx/lesson1_lifecycle/demo1/Demo1Activity.kt
package com.laioffer.my_mvx.lesson1_lifecycle.demo1

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.my_mvx.R

/**
 * @description: 版本1: 监听器监听生命周期
 * @date: 2022/9/19 7:52 上午
 */
class Demo1Activity : AppCompatActivity() {

  private var myListener: MyListener? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.empty_layout)

    myListener = MyListener()
  }

  override fun onStart() {
    super.onStart()
    myListener?.start()
  }

  override fun onStop() {
    super.onStop()
    myListener?.stop()
  }
}